library/core/src/num/nonzero.rs

```diff
@@ -2214,4 +2227,58 @@ mod verify {
     nonzero_check!(u64, core::num::NonZeroU64, nonzero_check_new_unchecked_for_u64);
     nonzero_check!(u128, core::num::NonZeroU128, nonzero_check_new_unchecked_for_u128);
     nonzero_check!(usize, core::num::NonZeroUsize, nonzero_check_new_unchecked_for_usize);
-}
+
+    macro_rules! nonzero_check_mul {
+        ($t:ty, $nonzero_type:ty, $nonzero_unchecked_mul:ident) => {
+            #[kani::proof_for_contract(NonZero::unchecked_mul)]
+            pub fn $nonzero_unchecked_mul_for() {
+                let x: NonZeroI8 = kani::any();
+                let y: NonZeroI8 = kani::any();
```

Comment:
We don't only want to check multiplication for `i8`s.
You should implement the [Arbitrary](https://model-checking.github.io/kani/crates/doc/kani/trait.Arbitrary.html) trait for each of the nonzero types (`NonZeroI8`, `NonZeroI16`), etc. See [this PR](https://github.com/model-checking/verify-rust-std/pull/136/files#diff-12e8ae8915eaa87c04610b5273f5e47589dea55cc44a7e40598b1593286523b6R1748) for an example. It would look something like:
```rust
impl Arbitrary for NonZeroI8 {
    fn any() -> NonZeroI8 {
      let x: i8 = kani::any();
      kani::assume(x != 0);
      NonZeroI8::new(x).unwrap()
    }
}
````
(You could even write a macro that generates these `Arbitrary` implementations for you).

Then your `proof_for_contract` macro could just do something like:
```rust
let x: $nonzero_ty = kani::any();
...
```
because your `Arbitrary` implementation ensures that each nonzero type has a `kani::any()` method.